connector/src/main/java/org/geysermc/connector/entity/AbstractArrowEntity.java

```diff
@@ -47,4 +47,12 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
 
         super.updateBedrockMetadata(entityMetadata, session);
     }
+
+    @Override
+    public Vector3f getBedrockRotation() {
+        double horizontalSpeed = Math.sqrt(motion.getX() * motion.getX() + motion.getZ() * motion.getZ());
```

Comment:
I'll have @RednedEpic double-check on this, but if I am correct square root operations are costly. Can we cache `yaw` and `pitch` used here and invalidate them on motion set?